File: 프로그래머스/lv2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
def prime(n):
    for i in range(2,int(n**0.5)+1):
        if n%i == 0:
            return False
    return True

def solution(numbers):
    cnt = 0
    numbers = list(numbers)
    arr = []
    for i in range(1,len(numbers)+1):
        arr+=list(permutations(numbers,i))
    arr1 = []
    for i in arr:
        arr1.append(int(''.join(i)))
    arr1 = list(set(arr1))
    for i in arr1:
        if i<=1:
            continue
        if prime(i):
            cnt+=1
    return cnt
            

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
function isPrime(number) {
    if(number===0) return false
    if(number===1) return false
    if(number===2) return true
    for(let i=2;i<=Math.sqrt(number);i++){
        if(number%i===0) return false
    }
    return true
}

function solution(numbers) {
    let prime=[]
    let nums=numbers.split("")
    let arr=[]
    let visited=Array(nums.length).fill(false)
    const dfs = (idx,arr) => {
        let n = Number(arr.join(""))
        if(!prime.includes(n) && isPrime(n)) prime.push(n)
        for(let i=0;i<nums.length;i++){
            if(visited[i]) continue
            visited[i]=true
            arr.push(nums[i])
            dfs(i,arr)
            visited[i]=false
            arr.pop()
        }
    }
    for(let i=0;i<nums.length;i++){
        visited[i]=true
        arr.push(nums[i])
        dfs(i,arr)
        visited[i]=false
        arr.pop(nums[i])
    }
    return prime.length;
}